client/web/reporter/src/views/Report/Builder.vue

```diff
@@ -883,17 +890,30 @@ export default {
       }
     },
 
+    updateEditorBlock (block = this.editor.block) {
+      this.reportBlocks[this.editor.currentIndex] = block
+      this.editor = undefined
+      this.refreshReport()
+    },
+
     editBlock (index = undefined) {
-      this.blocks.currentIndex = index
-      this.setCurrentDisplayElement(this.reportBlocks[this.blocks.currentIndex].elements.length ? 0 : undefined)
-      this.blocks.showConfigurator = true
+      this.editor = {
+        currentIndex: index,
+        block: JSON.parse(JSON.stringify(this.reportBlocks[index])),
```

Comment:
Since you stringify the block, it loses all class methods(same with display elements) so then they stop working. 
But if you map it back to a block class it will not work since the x, y, w, h properties are not there.
Therefore, make a new block and add the x, y, w, h and i properties to it.

Something like this:
```
const { x, y, w, h, i } = this.reportBlocks[index]
const block = new reporter.Block(this.reportBlocks[index])

block.x = x
block.y = y
block.w = w
block.h = h
block.i = i

this.editor = {
  currentIndex: index,
  block,
}
```